arangod/Aql/ExecutionNode/IndexNode.cpp

```diff
@@ -522,7 +522,9 @@ IndexNode::~IndexNode() = default;
 /// its unique dependency
 CostEstimate IndexNode::estimateCost() const {
   CostEstimate estimate = _dependencies.at(0)->getCost();
-  size_t incoming = estimate.estimatedNrItems;
+  size_t incoming = hasLimit()
```

Comment:
I think incoming should not be limited here.
`incoming` is the expected number of incoming items from the above node in the ExecutionPlan. This should not be affected by whether a limit is set for the node.
Instead, the number of _outgoing_ items should be affected by the limit.

Reply:
Aha ok, I missed that thanks.

Reply:
Is it possible to adjust the estimate in this part, because to adjust it it is not enough to just know the limit one must know against how many elements we are matching?

Reply:
Indeed! If understood your question correctly, then
```
FOR doc IN col
   FILTER doc.x IN ["A", "B", "C"]
   SORT doc.y
   LIMIT 10
   RETURN doc._key
```
The index node has to produce not 10, but 30 elements.